Fix minimum range difference for cols/rows in case of negative numbers

kaikun213 opened the pull request:
Currently a difference of -2 column/row steps will override a difference of 1 or 0.
This leads to wrong behavior e.g. in for fields that span multiple columns/rows.

Closes #413

kaikun213 committed:
Fix minimum range difference for columns and rows in case of negative numbers.

Currently a difference of -2 column/row steps will override a difference of 1 or 0.
This leads to wrong behaviour e.g. in  for fields that span multiple columns/rows.

Closes #413

lukehsiao reviewed (changes requested):
Can you add a test case that demonstrates what this fixes? You can create a new file in `tests/utils/test_utils_table.py`. 

**Update**: I went ahead an added the test case.

lukehsiao committed:
test(utils): add test for _min_range_diff

The purpose of this function's absolute parameter is:

    if absolute=True, return the absolute value of minimum magnitude difference
    if absolute=False, return the raw value of minimum magnitude difference

However, if absolute=False, we erroneously do NOT select the smallest
magnitude difference. For example,

    _min_range_diff(1, 2, 3, 4, absolute=False)

internally generates the list

    [-2, -3, -1, -2]

And rather than returning -1 (the minimum magnitude different), returns
-3. #420 fixes this bug.

lukehsiao committed:
docs(CHANGELOG): add minimum range difference fix

This also sorts the contributor names alphabetically.
